noir-projects/aztec-nr/aztec/src/context/avm.nr

```diff
@@ -80,9 +81,19 @@ impl AVMContext {
     #[oracle(avmOpcodeSendL2ToL1Msg)]
     pub fn send_l2_to_l1_msg(self, recipient: EthAddress, content: Field) {}
 
-    ///////////////////////////////////////////////////////////////////////////
-    // The functions below allow interface-equivalence with PrivateContext
-    ///////////////////////////////////////////////////////////////////////////
+    #[oracle(avmOpcodeCall)]
+    fn call<ARGS_COUNT, RET_SIZE>(
+        self,
+        gas: [Field; 3], // gas allocation: [l1Gas, l2Gas, daGas]
+        address: AztecAddress,
+        args: [Field; ARGS_COUNT],
+        temporary_function_selector: Field
```

Comment:
IDK why this one is called temporary. Calls will always be to a specific function, right?

Reply:
Yes, although there was an idea that the selector would just be included as the first field in calldata or something. 
Now i think of it it is nice to have separated. 
I will think more about this

Reply:
Coming back to this, if we can reduce the number of args on this array by 1 we should, we may need a register for each argument in the circuit representation. One less register is good!

Although we will have to think a bit more about keeping the private / public interfaces consistent

Reply:
Created https://github.com/AztecProtocol/aztec-packages/issues/5110 to track this.